control/controller.go

```diff
@@ -52,11 +58,18 @@ type Controller struct {
 }
 
 type Config struct {
-	// TODO(jsternberg): Integrate the concurrency and memory bytes quotas.
-	ConcurrencyQuota         int
-	MemoryBytesQuota         int64
+	// ConcurrencyQuota is the number of queries that are allowed to execute concurrently.
+	ConcurrencyQuota int
+	// MemoryBytesQuotaPerQuery is the maximum number of bytes (in table memory) a query is allowed to use at
+	// any given time.
+	//
+	// The maximum amount of memory the controller is allowed to consume is
+	//   ConcurrencyQuota * MemoryBytesQuotaPerQuery
 	MemoryBytesQuotaPerQuery int64
```

Comment:
Yeah, that's my understanding of how it should work long-term too.  Right now to address the issue if queryd OOMing, we just assume that each query will use the maximum amount of per-query memory, so we calculate the maximum memory in this simple way.

We have unit tests that verify that:
- A query will be terminated if it exceeds the memory limit
- Concurrent number of queries will be limited by the controller

So that we do have a very basic form of the hard limit on this branch.